test/astar.cc

```diff
@@ -1639,7 +1639,7 @@ TEST(BiDiAstar, test_recost_path) {
   vt::BidirectionalAStar astar;
 
   // hack hierarchy limits to allow to go through the shortcut
-  mode_costing[int(travel_mode)]->RelaxHierarchyLimits(0.f, 0.f);
+  mode_costing[int(travel_mode)]->RelaxHierarchyLimits(true);
```

Comment:
@kevinkreiser I'm not sure how to cope with this hack.

Reply:
The change to the hack above leads to the test failure 
https://app.circleci.com/pipelines/github/valhalla/valhalla/7514/workflows/08681a6b-9258-4295-b132-2a7439addc8a/jobs/46371

```
021/08/09 09:21:15.482272 [TRACE] /root/project/src/thor/bidirectional_astar.cc:1012: CONNECTIONS FOUND 2
[[0.031441,0.000000],[0.085340,0.000000]],
[[0.022458,-0.008983],[0.031441,0.000000]],
2021/08/09 09:21:15.482290 [DEBUG] path_cost::898
2021/08/09 09:21:15.482303 [DEBUG] FormPath path_iterations::6,8
/root/project/test/astar.cc:1653: Failure
Expected equality of these values:
  path.size()
    Which is: 5
  6
Final path has wrong number of edges
[  FAILED  ] BiDiAstar.test_recost_path (1354 ms)
[----------] 1 test from BiDiAstar (1354 ms total)

[----------] Global test environment tear-down
[==========] 36 tests from 3 test suites ran. (4212 ms total)
[  PASSED  ] 35 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] BiDiAstar.test_recost_path

 1 FAILED TEST
```

Reply:
it doesnt look easy unfortunately. seems like you'd have to clone auto costing into this compilation unit, make its hierarchy limits accessible, and then register its create function with the costfactory. then you'd have to cast it to its cloned class name and call the accessible hierarchy limits function you added. its doable, but a giant mess tbh. what makes it a pain is that the costing classes are all pimpls... if we could include the source file with the definition of autocost into this test as a header (ie if it were a private header) then you could just derive from autocost, and add a new function called hackHierarchyLimits, register that costing type and move on... i think

let me know if that helps in anyway, in the meantime ill give it some more thought. sorryy for the annoyance

Reply:
> if we could include the source file with the definition of autocost into this test as a header (ie if it were a private header) then you could just derive from autocost, and add a new function called hackHierarchyLimits, register that costing type and move on...

Actually, I've tried something like that and it works.

Then, I've realised the hierarchy limits can be read-write accessed too, so I'm now trying simpler solution, see https://github.com/valhalla/valhalla/pull/3253/commits/f136f226070218738925a95700f05843de07ae43